Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md

```diff
@@ -166,7 +166,84 @@ pyautogui.doubleClick()
 ![sample_gif](./assets/move_folder.gif)
 
 ## Keyboard Controls
-Pending
+## Keyboard Controls
+Once familiar with mouse functions, exploring keyboard interactions becomes the next vital step. PyAutoGui offers various methods for keyboard input, enhancing automation capabilities.
+
+We can write something using the `write()` keyword. This will simply type the argument that is passed into it, which is a string.
+
+```python
+>>> pyautogui.write('Hi!')                  # Prints 'Hi'
+>>> pyautogui.write('Hi!', interval=0.25)   # Similar to the move function, the second argument is
```

Comment:
Great example, Mann! Could you please add some pictures/gifs to demonstrate an example? Another addition to consider is keyboard presses while the program is running. For example, what will happen if I press the caps lock button on my keyboard mid-execution?

Reply:
Sure, will add visuals to support the tutorial.